### ISSUE TITLE
Cannot dynamically adjust Facets within a SearchProvider

### ISSUE THREAD

**Issue by alexanderatallah** [elastic/search-ui#464]
It looks like once a `SearchProvider` has been initialized, there's no way to adjust the `facets` part of the search `Context`. Am I right about this limitation?

This means that when we have React components dynamically change the aggregations needed in our Elasticsearch queries, we can't change the search `Context`'s `facets` and use them to build a request.

Here's the code I'd like to execute in place of your Elasticsearch example: https://github.com/elastic/search-ui/blob/master/examples/elasticsearch/src/buildRequest.js#L43

```
export const buildRequest = (state: Partial<Context>) => {
  const {
    current,
    filters,
    facets,
    resultsPerPage,
    searchTerm,
    sortDirection,
    sortField
  } = state

  const aggs = buildAggregations(facets)
...
```
But currently, since I start with no faceting, `facets` is always `undefined` unless a child component can re-configure the SearchProvider on the top level.

Best solution would be if we could have `Facet` components encapsulate the configuration of facets, changing the next query that's executed. Then, parent components don't have to worry about the data dependencies of their children.

**Comment by JasonStoltz** [comment]
Generally speaking that is true, yes.

However, if you are using Elasticsearch as your back end this is not true.

Since there is no connector available for Elasticsearch, it is on you to write the logic to determine which facets or aggregations are fetched. That means you can apply whatever logic you wish here https://github.com/elastic/search-ui/blob/master/examples/elasticsearch/src/buildRequest.js#L43.

Does that make sense?

**Comment by JasonStoltz** [comment]
@alexanderatallah A fix for this is available in 1.4.0-rc.0. If you get a chance to test it out, let me know how it works for you.
